tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -751,3 +751,10 @@ function funcAcceptingIntArgs(int... scores) returns int {
     };
     return fn();
 }
+
+function errorConstructorReferenceTest() returns error? {
+    string temp = "text";
+    var process = function() returns error? {
+        return error("An Error", tempString = temp);
```

Comment:
Let's also use a module-level variable as another detail argument.

After this line, you can even change one of the variables and assert that the change is reflected where the function is actually called.